src/core/tests/mod.rs

```diff
@@ -0,0 +1,1764 @@
+
+                // should succeed
+                MemPoolDB::try_add_tx(
+                    &mut mempool_tx,
+                    &mut chainstate,
+                    &ConsensusHash([0x1 + (block_height as u8); 20]),
+                    &BlockHeaderHash([0x2 + (block_height as u8); 32]),
+                    txid.clone(),
+                    tx_bytes,
+                    tx_fee,
+                    block_height as u64,
+                    &origin_addr,
+                    origin_nonce,
+                    &sponsor_addr,
+                    sponsor_nonce,
+                    None,
+                )
+                .unwrap();
+
+                txids.push(txid);
+            }
+            mempool_tx.commit().unwrap();
+
+            let ts_1 = get_epoch_time_ms();
+            let ms = mempool.make_mempool_sync_data().unwrap();
+            let ts_2 = get_epoch_time_ms();
+            eprintln!(
+                "make_mempool_sync_data({}): {} ms",
+                txids.len(),
+                ts_2.saturating_sub(ts_1)
+            );
+
+            let mut present_count: u32 = 0;
+            let mut absent_count: u32 = 0;
+            let mut fp_count: u32 = 0;
+            match ms {
+                MemPoolSyncData::BloomFilter(ref bf) => {
+                    eprintln!(
+                        "bloomfilter({}); txids.len() == {}",
+                        block_height,
+                        txids.len()
+                    );
+                    let recent_txids = mempool.get_bloom_txids().unwrap();
+                    assert!(recent_txids.len() <= MAX_BLOOM_COUNTER_TXS as usize);
+
+                    let max_height = MemPoolDB::get_max_height(mempool.conn())
+                        .unwrap()
+                        .unwrap_or(0);
+                    eprintln!(
+                        "bloomfilter({}): recent_txids.len() == {}, max height is {}",
+                        block_height,
+                        recent_txids.len(),
+                        max_height
+                    );
+
+                    let mut recent_set = HashSet::new();
+                    let mut in_bf = 0;
+                    for txid in recent_txids.iter() {
+                        if bf.contains_raw(&txid.0) {
+                            in_bf += 1;
+                        }
+                        recent_set.insert(txid.clone());
+                    }
+
+                    eprintln!("in bloom filter: {}", in_bf);
+                    assert!(in_bf >= recent_txids.len());
+
+                    for txid in txids.iter() {
+                        if !recent_set.contains(&txid) && bf.contains_raw(&txid.0) {
+                            fp_count += 1;
+                        }
+                        if bf.contains_raw(&txid.0) {
+                            present_count += 1;
+                        } else {
+                            absent_count += 1;
+                        }
+                    }
```

Comment:
This code path is verifying that the bloom filter's false positive rate is sufficiently small.  This does not apply to `TxTags`, since the `TxTags` query variant does not use a bloom filter.